### PR TITLE
Allow banning clients not connected to the server

### LINKED ISSUE #96: Allow banning players not currently connected to the server
Currently, Akashi only allows banning a client if that client is currently connected. You cannot, for example, ban an IPID that is not currently connected. This should be changed to allow banning any IPID or HDID, even if that IPID is not currently connected to the server.

### PR DESCRIPTION
**Pull request by MangosArentLiterature**
Allows you to ban any IPID, regardless of whether it is connected to the server or not. To accomplish this, IPIDs are used for IP bans instead of remote IPs.

In addition, this PR cleans up both isIPBanned() and isHDIDBanned() by removing extraneous DB queries. It removes getBanDuration and getBanReason, and instead isIPBanned and isHDIDBanned() will retrieve this info in it's single query.

Closes #96

### CODE REVIEW

**Review by in1tiate — APPROVED** [review]
looks gucci

File: core/include/db_manager.h [hunk 1]
```diff
@@ -56,52 +56,26 @@ class DBManager{
...
-     * @param ip The IP address to check if it is banned.
+     * @param ipid The IPID to check if it is banned.
      *
-     * @return True if the query could return at least one such record.
+     * @return A pair of values:
+     * * First, a `bool` that is true if the query could return at least one such record.
+     * * Then, a `QString` that is the reason for the ban.
      */
-    bool isIPBanned(QHostAddress ip);
+    std::pair<bool, QString> isIPBanned(QString ipid);
```
Comment (scatterflower) [inline core/include/db_manager.h:67]: Use a QPair

File: core/src/db_manager.cpp [hunk 2]
```diff
@@ -31,101 +31,46 @@ DBManager::DBManager() :
         updateDB(db_version);
 }
 
-bool DBManager::isIPBanned(QHostAddress ip)
+std::pair<bool, QString> DBManager::isIPBanned(QString ipid)
```
Comment (scatterflower) [inline core/src/db_manager.cpp:34]: QPair

File: core/src/db_manager.cpp [hunk 3]
```diff
@@ -31,101 +31,46 @@ DBManager::DBManager() :
...
 {
     QSqlQuery query;
-    query.prepare("SELECT TIME FROM BANS WHERE IP = ? ORDER BY TIME DESC");
-    query.addBindValue(ip.toString());
+    query.prepare("SELECT * FROM BANS WHERE IPID = ? ORDER BY TIME DESC");
```
Comment (scatterflower) [inline core/src/db_manager.cpp:37]: instead of selecting *, only select the fields you need

File: core/src/db_manager.cpp [hunk 4]
```diff
@@ -31,101 +31,46 @@ DBManager::DBManager() :
...
-        else return false;
+            return {true, reason};
+        else return {false, nullptr};
     }
-    else return false;
+    else return {false, nullptr};
 }
 
-bool DBManager::isHDIDBanned(QString hdid)
+std::pair<bool, QString> DBManager::isHDIDBanned(QString hdid)
```
Comment (scatterflower) [inline core/src/db_manager.cpp:54]: QPair

File: core/src/db_manager.cpp [hunk 5]
```diff
@@ -31,101 +31,46 @@ DBManager::DBManager() :
...
 {
     QSqlQuery query;
-    query.prepare("SELECT TIME FROM BANS WHERE HDID = ? ORDER BY TIME DESC");
+    query.prepare("SELECT * FROM BANS WHERE HDID = ? ORDER BY TIME DESC");
```
Comment (scatterflower) [inline core/src/db_manager.cpp:57]: Ditto previous comment about selecting *

**Review by scatterflower — CHANGES_REQUESTED** [review]

**Pull request merged**